aiida_wannier90/calculations.py

```diff
@@ -187,14 +195,48 @@ def _SEEDNAME(self):
         Return the default seedname, unless a custom one has been set in the
         calculation settings
         """
-        return self.inputs.metadata.options.seedname
+        input_filename = self.inputs.metadata.options.input_filename
+
+        if input_filename.endswith(self._REQUIRED_INPUT_SUFFIX):
```

Comment:
If I understand this correctly, the second ``return`` should never be hit, right? 

Since that is essentially "dead code", we should turn the ``if`` into an ``assert`` (checks our assumptions about the program), and then  ``return input_filename[:-len(self._REQUIRED_INPUT_SUFFIX)]`` unconditionally.

Reply:
I am not sure - the validation is currently done only in the `prepare_for_submission`. So it is possible (at least in the engine) to have an instance of the `Process` class where this condition is not met and the second `return` is reached.

As I didn't like throwing exceptions in a property, I thought it was nicer to do it this way - while in the parser I instead raise: 
https://github.com/aiidateam/aiida-wannier90/pull/42/files/96047c7edbe4f44de32590d31bf4c8b434558463#diff-688d87b07e812b5cec1621f6d71ea539R36-R51

If you feel it's better to raise anyway, I can raise instead of the second return statement (I agree this should, in practice, not happen).

Reply:
Yeah I think I'd prefer the raise, but don't feel _very_ strongly about it.

Reply:
Ok, changed to a `raise`.